packages/docusaurus-plugin-content-blog/src/blogUtils.ts

```diff
@@ -363,6 +364,65 @@ async function processBlogSourceFile(
   };
 }
 
+type SortBlogPostsFn = (args: {blogPosts: BlogPost[]}) => void | BlogPost[];
+
+type SortBlogPostsPreset = 'ascending' | 'descending';
+
+// eslint-disable-next-line @typescript-eslint/consistent-indexed-object-style
+const sortPresets: Record<SortBlogPostsPreset, SortBlogPostsFn> = {
+  descending: ({blogPosts}) =>
+    blogPosts.sort(
+      (a, b) => b.metadata.date.getTime() - a.metadata.date.getTime(),
+    ),
+  ascending: ({blogPosts}) =>
+    blogPosts.sort(
+      (a, b) => a.metadata.date.getTime() - b.metadata.date.getTime(),
+    ),
+};
+
+interface SortBlogPostsOptions {
+  blogPosts: BlogPost[];
+  sortPosts: 'ascending' | 'descending' | SortBlogPostsFn;
+}
+
+function getSortFunction(sortPosts: Options['sortPosts']): SortBlogPostsFn {
+  if (typeof sortPosts === 'function') {
+    return sortPosts;
+  }
+
+  if (typeof sortPosts === 'string') {
+    const presetFn = sortPresets[sortPosts];
+    if (!presetFn) {
+      throw new Error(
+        `sortPosts preset ${sortPosts} does not exist, valid presets are: ${Object.keys(
+          sortPresets,
+        ).join(', ')}`,
+      );
+    }
+    return presetFn;
+  }
+
+  return () => [];
+}
+
+function sortBlogPosts({
+  blogPosts,
+  sortPosts,
+}: SortBlogPostsOptions): BlogPost[] {
+  const sortFunction = getSortFunction(sortPosts);
+  const sortedBlogPosts = sortFunction({blogPosts});
+
+  if (sortedBlogPosts !== undefined) {
+    return sortedBlogPosts;
+  }
+
+  throw new Error(
+    `Invalid blog config sortPost value: ${sortPosts}, must be one of: ${Object.keys(
+      sortPresets,
+    ).join(', ')}`,
+  );
```

Comment:
no need to throw here IMHO we already throw inside `getSortFunction`

The user is allowed to provide a sort function that returns undefined, and this error prevents that kind of usage.